src/telegram.cpp

```diff
@@ -613,7 +613,19 @@ uint16_t TxService::read_next_tx(uint8_t offset) {
     if (telegram_last_->offset != offset) {
         return 0;
     }
-    add(Telegram::Operation::TX_READ, telegram_last_->dest, telegram_last_->type_id, telegram_last_->offset + 25, message_data, 1, 0, true);
+
+    uint8_t add_offset;
+    if (telegram_last_->dest == 0x08 && telegram_last_->type_id == 0xC2) {  //fix for overlapping 0xC2 telegrams from Boiler. Are other telegrams also affected?
```

Comment:
> Are other telegrams also affected

@pswid  You're asking the right question. The answer is yes, all ems1.0 telegrams have 5 byte header and 27 data, ems+ telegrams 7 byte header and 25 byte data. Emsesp adds 0x100 to all ems+ telegrams, so you can check for type_id > 0xFF to decide what offset is needed. The overflow seems special to the C2 telegram, i have not seen this before, but my boiler sends only 26 bytes for 0x10 telegram with offset 0, but the telegram is much longer. You can check my implementation [here](https://github.com/MichaelDvP/EMS-ESP32/blob/dev2/src/telegram.cpp#L610-L624)

Reply:
My solution was just a simple fix, yours is comprehensive and more professional. However, in your implementation, the condition checking if the date is valid (in boiler.cpp):

    // check for valid date, https://github.com/emsesp/EMS-ESP32/issues/204
    if (telegram->message_data[11] & 0x80) {

is wrong, because in my case it causes the error to be displayed in the form:
![image](https://user-images.githubusercontent.com/78219494/145271238-6abaf33e-2e4e-49f4-96f5-2dcfd07327c8.png)
instead of:
![image](https://user-images.githubusercontent.com/78219494/145271361-41741c24-c514-4de1-9c07-888519d485de.png)
BTW, some units are displayed incorrectly in your version (e.g. uA instead of minutes) see above.

There are also other telegrams, the reading of which caused the offset to overflow, e.g. 0xC6 & 0xC7 and probably some others.

Reply:
Thanks, it should be `message_data[10]` to alllow #121 and #204. The units should work after page reload (browser cache).
But my personal build is a permanent alpha, no releases, and dont care about breaking changes. I only wanted to show another solution and clearify the telegram data length.  
I first see the overflow in #228 for the 0xC2 (my boiler does not have it), but i agree that if there is one telegram with this behaviour, probably there are others, and  overflow should always be checked.

Reply:
You're right. After page reload units are OK. 
BTW, your personal build visually looks very good. I would even say it's nicer than the official version.

Reply:
> You're right. After page reload units are OK. BTW, your personal build visually looks very good. I would even say it's nicer than the official version.

Looks like Michael is using my latest work from the ft_reacthooks branch. sneaky! It's not public yet, still adding some features but I plan to push it to dev in the next few days.

Reply:
> Looks like Michael is using my latest work from the ft_reacthooks branch.

That's right, I like the new look and as proposed the code is better readable and easier to enhance.  
> sneaky! It's not public yet,

That's a lie. The branch (https://github.com/emsesp/EMS-ESP32/tree/ft_reacthooks) is public and you ask [here](https://github.com/emsesp/EMS-ESP32/issues/124#issuecomment-985022031) for testing it. Now you call it sneaky if someone do a test? Strange!

Reply:
@MichaelDvP no need to be rude. What ever is bothering you let me know so we can continue to work together. You're are admin on this project and have made some great contributions, and I enjoy working together, but your recent attitude is concerning. Great you merged ft_reacthooks into your dev branch - any feedback on the new UI so far?